src/mail.sh

```diff
@@ -357,6 +362,94 @@ function find_commit_references()
   return 22 # EINVAL
 }
 
+# This function check over the kw mail groups options and
+# calls the proper handling function for every option.
+#
+# Returns:
+# returns 0 if successful, non-zero otherwise
+function kw_mail_groups()
+{
+  if [[ -n "${options_values['GROUPS_CREATE']}" ]]; then
+    create_kw_mail_group "${options_values['GROUP']}"
+    return "$?"
+  fi
+
+  return 0
+}
+
+# This function creates a new kw mail group
+#
+# @group_name: The name of the group that will be created.
+#
+# Returns:
+# returns 0 if suc cessful, non-zero otherwise
+function create_kw_mail_group()
+{
+  local group_name="$1"
+  local values
+
+  validate_group_name "$group_name" || exit_msg 'Error while creating new group'
```

Comment:
I recommend using the traditional if command here, also, avoid exit here and just return some error code. The same idea applies to the below line.